internal/kafka/internal/services/kafka.go

```diff
@@ -1342,3 +1347,12 @@ func (k *kafkaService) UpdateZeroValueOfKafkaRequestsExpiredAt() error {
 
 	return db.Error
 }
+
+func (k *kafkaService) IsSubscriptionActive(kafkaRequest *dbapi.KafkaRequest) (*bool, error) {
```

Comment:
Why `*bool`? I believe this should be 
```suggestion
func (k *kafkaService) IsSubscriptionActive(kafkaRequest *dbapi.KafkaRequest) (bool, error) {
```

In case of error we return the `zero value` for `bool` (false) and the error

Reply:
I was trying to be cautious in case someone doesn't check the error and to avoid `false` having two meanings. If an error is returned, `false` here could be a false negative, it may not mean that the subscription is actually expired.

Reply:
+1 to change it to `bool, error`.
In Go in general it is assumed that if an error is returned the value of the result is not valid / reliable unless explicitly stated.